Documentation/Input/InputEvents.md

```diff
@@ -1,23 +1,169 @@
+| [`IMixedRealityInputHandler`](xref:Microsoft.MixedReality.Toolkit.Input.IMixedRealityInputHandler) | Input Down / Up | Raised on changes to binary inputs like buttons. |
+| [`IMixedRealityInputHandler<T>`](xref:Microsoft.MixedReality.Toolkit.Input.IMixedRealityInputHandler`1) | Input Changed | Raised on changes to inputs of the given type. **T** can take the following values: <br/> - *float* (e.g returns analog trigger)<br/> - *Vector2* (e.g returns gamepad thumbstick direction) <br/> - *Vector3* (e.g return position of tracked device) <br/> - *Quaternion* (e.g returns orientation of tracked device)<br/> - [MixedRealityPose](xref:Microsoft.MixedReality.Toolkit.Utilities.MixedRealityPose) (e.g. returns fully tracked device) |
+| [`IMixedRealitySpeechHandler`](xref:Microsoft.MixedReality.Toolkit.Input.IMixedRealitySpeechHandler) | Speech Keyword Recognized | Raised on recognition of one of the keywords configured in the *Speech Commands Profile*. |
+| [`IMixedRealityDictationHandler`](xref:Microsoft.MixedReality.Toolkit.Input.IMixedRealityDictationHandler) | Dictation<br/> Hypothesis <br/> Result <br/> Complete <br/> Error | Raised by dictation systems to report the results of a dictation session. |
+| [`IMixedRealityGestureHandler`](xref:Microsoft.MixedReality.Toolkit.Input.IMixedRealityGestureHandler) | Gesture events on: <br/> Started <br/> Updated <br/> Completed <br/> Canceled | Raised on gesture detection. |
+| [`IMixedRealityGestureHandler<T>`](xref:Microsoft.MixedReality.Toolkit.Input.IMixedRealityGestureHandler`1) | Gesture Updated / Completed | Raised on detection of gestures containing additional data of the given type. See [**Gesture Events**](Gestures.md#gesture-events) for details on possible values for **T**. |
+| [`IMixedRealityHandJointHandler`](xref:Microsoft.MixedReality.Toolkit.Input.IMixedRealityHandJointHandler) | Hand Joints Updated | Raised by articulated hand controllers when hand joints are updated. |
+| [`IMixedRealityHandMeshHandler`](xref:Microsoft.MixedReality.Toolkit.Input.IMixedRealityHandMeshHandler) | Hand Mesh Updated | Raised by articulated hand controllers when a hand mesh is updated. |
+| [`IMixedRealityInputActionHandler`](xref:Microsoft.MixedReality.Toolkit.Input.IMixedRealityInputActionHandler) | Action Started / Ended | Raise to indicate action start and end for inputs mapped to actions. |
+
+## Input events in action
+
+At the script level, input events can be consumed by implementing one of the event handler interfaces shown in the table above. When an input event fires via an user interaction, the following takes place:
+
+1. The MRTK input system recognizes that an input event has occurred
+1. The MRTK input system fires the relevant interface function of the input event to all registered global input handlers
```

Comment:
"registered global input handlers" consider adding a link to the section below on global events)